input/docs/handbook/collections/index.md

```diff
@@ -82,14 +82,124 @@ var myBindingOperation = mySource
+If you have a unique id, you should use an observable cache as it is dictionary based which will ensure no duplicates can be added and it notifies on adds, updates and removes, whereas list allows duplicates and only has no concept of an update.
 
 There is another difference. The cache side of dynamic data is much more mature and has a wider range of operators. Having more operators is mainly because I found it easier to achieve good all round performance with the key based operators and do not want to add anything to Dynamic Data which inherently has poor performance.
 
-## How to track changes in collections of Reactive Objects?
+# Using DynamicData with ReactiveUI
+
+When building applications with ReactiveUI and DynamicData, you have a choice to work with mutable or with immutable collections. When working with immutable ones, using an `ObservableAsPropertyHelper<T>` is enough in simple cases. The `ObservableAsPropertyHelper<T>` represents an `Observable<T>`, a stream of values over time. You can treat those values as events, and the new values as event arguments. This means if you are using immutable collections, you can treat them as event arguments and update a property with a new collection each time it changes.
+
+### Simple Scenario
+
+```cs
+public class SimpleService 
+{
+    private readonly Subject<IEnumerable<bool>> _items = new Subject<IEnumerable<bool>>();
+
+    public IObservable<IEnumerable<bool>> Items => _items;
+
+    public Service() => _items.OnNext(new List<bool> { true, false, true, false });
+}
+```
+
+In this example we have a service that publishes a new collection each time the data set changes. If you subscribe to the `IObservable<IEnumerable<bool>>` exposed by the `SimpleService`, you will receive a new collection for each update. This is the most simple approach, and, for sure, it's less performant on larger data sets.
+
+```cs
+public class ViewModel : ReactiveObject
+{
+    private readonly ObservableAsPropertyHelper<IEnumerable<bool>> _items;
+    public IEnumerable<bool> Items => _items.Value;
+
+    public ViewModel()
+    {
+        var service = new SimpleService();
+        _items = service.Items
+            // Transform the incoming new collections as you wish
+            // using the power of reactive extensions.
+            .Select(items => items.Skip(1))
+            // And finally convert the collection to a property,
+            // declared as ObservableAsPropertyHelper. Remember, 
+            // that for each change a *new* collection is published.
+            // Don't try to mutate it, this won't work. Read further
+            // to learn how to work with mutable collections.
+            .ToProperty(this, x => x.Items);
+    }
+}
+```
+
+### Complex Scenario
+
+```cs
+public class ComplexService 
+{
+    public SourceList<bool> Items { get; } = new SourceList<bool>();
```

Comment:
A better approach would be to expose a `IObservable<IChangeSet<bool>> GetItemChangeSet() =>`. You're not using anything on the Items to justify leaving it immutable and allows you to use `Observable.Publish` and recount

Reply:
@glennawatson i think Items should be a collection of domain objects, not a stream of changes. In case of immutability replacing SourceList<bool> with IObservableList<bool> make sense.

Reply:
It's one option to go but I don't agree in this case. Just due to the usage pattern of then consuming elsewhere with Connect(). You still have the inheritant performance issue of regenerating your connect() every run. 

We use the idiom of abstracting a lot of controls via the base condition `IObservable<IChangeSet<bool>> ` in the rxui website to avoid that hit.

Reply:
@chuuddo I think I would agree with your point if we didn't care about the data actually being a stream of observable changes. I found the `IObservableList` a bit deceptive to new users of the DynamicData framework, because they try to pass list all the way down their hierarchy for binding, when there is only really two common use cases, Bind() to expose data to the world, or dealing with observable change sets.